### PR TITLE
feat(metric): custom slot to render contents in gap

### LINKED ISSUE #2182: Add render slot for gap area in Metric
**Is your feature request related to a problem? Please describe.**
Solutions teams have requested a way to consistently place tags below the title on the Metric component. This is not easy to force externally with consistency.

image

**Describe the solution you'd like**
Ideally we allow a render prop on the `Metric` chart that we pass each datum to in order to render the contents into the gap area.

image

The user would be responsible for padding and sizing of the gap content as applicable to their use case.

**Describe alternatives you've considered**
Rendering custom elements absolutely relative to the `Metric` chart. Causes inconsistent placement of contents.

**Additional context**
Since the `Metric` uses dom elements this approach is preferred over dimensional placement of the render contents.

### PR DESCRIPTION
**Pull request by nickofthyme**
## Summary

This PR adds the ability to render arbitrary contents into the gap space of the metric. This adds the optional `slot` prop to the `Metric` spec.

Screen Recording 2024-01-16 at 01 06 53 PM

## Details

The contents of the slot are contained inside of an absolutely positioned container to allow rendering contents without contents size impacting the layout of the Metric elements. Alternatively, we could add an option to disable this behavior ad allow the Metric elements to be pushed out of the render area or we could account for the slot size at the time of rendering.

## Issues

closes #2182

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated
- [x] The code has been checked for cross-browser compatibility (Chrome, Firefox, Safari, Edge)
- [x] Visual changes have been tested with `light` and `dark` themes

### CODE REVIEW

**Comment by nickofthyme** [comment]
buildkite update screenshots

File: packages/charts/api/charts.api.md [hunk 1]
```diff
@@ -1987,6 +1987,8 @@ export interface MetricSpec extends Spec {
     // (undocumented)
     data: (MetricDatum | undefined)[][];
     // (undocumented)
+    slot?: ReactNode;
```
Comment (markov00) [inline packages/charts/api/charts.api.md:1990]: 🦥 do we have a better name for that? isn't too generic?

**Review by markov00 — APPROVED** [review]
I'm fine with the code changes, just the naming is a bit too vague

File: packages/charts/api/charts.api.md (hunk 1 above)
Comment (nickofthyme) [reply packages/charts/api/charts.api.md:1990]: Yeah what are you thinking?

Comment (markov00) [reply packages/charts/api/charts.api.md:1990]: I don't know, could be something like `content` or `body`

Comment (nickofthyme) [reply packages/charts/api/charts.api.md:1995]: Renamed to `body` in 5156195

**Comment by nickofthyme** [comment]
buildkite update screenshots

**Pull request merged**